tests/rptest/tests/adjacent_segment_merging_test.py

```diff
@@ -0,0 +1,123 @@
+        )
+
+        self.s3_bucket_name = si_settings.cloud_storage_bucket
+
+        super(AdjacentSegmentMergingTest,
+              self).__init__(test_context=test_context,
+                             extra_rp_conf=xtra_conf,
+                             si_settings=si_settings)
+
+        self.kafka_tools = KafkaCliTools(self.redpanda)
+        self.rpk = RpkTool(self.redpanda)
+
+    def setUp(self):
+        super().setUp()  # topic is created here
+
+    def tearDown(self):
+        self.s3_client.empty_bucket(self.s3_bucket_name)
+        super().tearDown()
+
+    @cluster(num_nodes=3)
+    @matrix(acks=[-1, 0, 1])
+    def test_reupload_of_local_segments(self, acks):
+        """Test adjacent segment merging using using local data.
+        The test starts by uploading large number of very small segments.
+        The total amount of data produced is smaller than the target segment
+        size. Because of that, after the housekeeping we should end up with
+        only one segment in the cloud.
+        The retention is not enable so the reupload process can use data 
+        available locally.
+        """
+        for _ in range(0, 10):
+            # Every 'produce' call should create at least one segment
+            # in the cloud which is 1MiB
+            self.kafka_tools.produce(self.topic, 1024, 1024, acks)
+            time.sleep(1)
+        time.sleep(5)
+
+        def manifest_has_one_segment():
+            try:
+                num_good = 0
+                for path in self._find_partition_manifests():
+                    manifest = self._download_partition_manifest(path)
+                    target_lower_bound = 1024 * 1024 * 8
+                    for name, meta in manifest["segments"].items():
+                        self.logger.info(
+                            f"segment {name}, segment_meta: {meta}")
+                        if meta["size_bytes"] >= target_lower_bound:
+                            # we will only see large segments with size
+                            # greater than lower bound if housekeeping
+                            # is working
+                            num_good += 1
+                return num_good > 0
+            except:
+                return False
+
+        wait_until(manifest_has_one_segment, 60)
+
+    def _find_partition_manifests(self):
+        res = []
+        for obj in self.s3_client.list_objects(self.s3_bucket_name):
+            if obj.Key.endswith("manifest.json") and not obj.Key.endswith(
+                    "topic_manifest.json"):
+                res.append(obj.Key)
+        return res
+
+    def _download_partition_manifest(self, manifest_path):
+        """Find and download individual partition manifest"""
+        manifest = self.s3_client.get_object_data(self.s3_bucket_name,
+                                                  manifest_path)
+        self.logger.info(f"manifest found: {manifest}")
+        return json.loads(manifest)
```

Comment:
Rename `s3_client` to `cloud_storage_client` and use `key` instead of `Key`.
See https://github.com/redpanda-data/redpanda/pull/8086/commits/46d5495b5392d2601c583a47e14786ebbcfc8229 and https://github.com/redpanda-data/redpanda/pull/8086/commits/011363d7024a02b049c57bc6ea063db47a7c25af.